tests/utils.rs

```diff
@@ -0,0 +1,102 @@
+use std::{process::Stdio, sync::Arc};
+
+use anyhow::Result;
+use log::info;
+use neo4rs::query;
+use pubky_nexus::{
+    db::connectors::redis::get_redis_conn, get_neo4j_graph, redis_is_empty, reindex, routes, setup,
+    Config,
+};
+use tokio::{
+    net::TcpListener,
+    sync::{Mutex, OnceCell},
+};
+
+// Util backend server for testing
+// Performs the same routine the main service server does
+// OnceCell is used to ensure the server is only started once
+#[derive(Clone, Debug)]
+pub struct TestServiceServer {
+    pub initialized: bool,
+}
+
+pub static TEST_SERVER: OnceCell<Arc<Mutex<TestServiceServer>>> = OnceCell::const_new();
+
+impl TestServiceServer {
+    pub async fn get_test_server() -> Arc<Mutex<TestServiceServer>> {
+        // Start the server if it hasn't been started
+        TEST_SERVER
+            .get_or_init(|| async {
+                Self::start_server().await.unwrap();
+                Arc::new(Mutex::new(TestServiceServer { initialized: true }))
+            })
+            .await
+            .to_owned()
+    }
+
+    async fn sync_graph() {
+        let graph = get_neo4j_graph().expect("Failed to get Neo4j graph connection");
+
+        // drop and run the queries again
+        let drop_all_query = query("MATCH (n) DETACH DELETE n;");
+        graph
+            .lock()
+            .await
+            .run(drop_all_query)
+            .await
+            .expect("Could not drop graph nodes.");
+
+        // Run the run-queries.sh script on the Docker host using docker exec
+        tokio::process::Command::new("docker")
+            .args(&["exec", "neo4j", "bash", "/db-graph/run-queries.sh"])
```

Comment:
There is a few trade offs with this approach as:
- We don't know where neo4j is (maybe not docker, maybe different containter name (as in `pubky-docker`)
- Very long waits when you simply want to run a single test case often during development

Is there a way we can run only `sync_graph()` optionally? I am thinking maybe we can have a more granular control envvar than `SYNC_DB`, maybe an explicit `SYNC_GRAPH` variable ? We already know whether to reindex or not with the `REINDEX` envvar, this way we control both independently.

Reply:
Will see what I can do.  
About your first point, that's very true. It was just a ton easier to do it this way for now. After all, it's for local tests and pipelines which won't be changing often.